qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -0,0 +1,368 @@
+                "'{}'".format(type(other).__name__)
+            )
+
+        # Check compatibility (i.e. operators act on same register length)
+        if self.register_length != other.register_length:
+            raise TypeError("Incompatible register lengths for '+'.")
+
+        label1, coeffs1 = zip(*self.to_list())
+        label2, coeffs2 = zip(*other.to_list())
+
+        return FermionicOp(list(zip(label1 + label2, coeffs1 + coeffs2)))
+
+    def to_list(self) -> List[Tuple[str, complex]]:
+        """Getter for the operator_list of `self`"""
+        return list(zip(self._labels, self._coeffs))
+
+    @property
+    def register_length(self) -> int:
+        """Getter for the length of the fermionic register that the FermionicOp `self` acts
+        on.
+        """
+        return self._register_length
+
+    def adjoint(self) -> "FermionicOp":
+        """Returns the complex conjugate transpose (dagger) of `self`."""
+
+        dagger_map = {"+": "-", "-": "+", "I": "I", "N": "N", "E": "E"}
+        label_list = []
+        coeff_list = []
+        for label, coeff in zip(self._labels, self._coeffs):
+            conjugated_coeff = coeff.conjugate()
+
+            daggered_label = ""
+
+            for i, char in enumerate(label):
+                daggered_label += dagger_map[char]
+                if char in ["+", "-"]:
+                    permute_through = label[i + 1:]
+                    conjugated_coeff *= (-1) ** (
+                        permute_through.count("+") + permute_through.count("-")
+                    )
+
+            label_list.append(daggered_label)
+            coeff_list.append(conjugated_coeff)
+
+        return FermionicOp(list(zip(label_list, coeff_list)))
+
+    def reduce(self, atol: Optional[float] = None, rtol: Optional[float] = None) -> "FermionicOp":
+        if atol is None:
+            atol = self.atol
+        if rtol is None:
+            rtol = self.rtol
+
+        label_list, indexes = np.unique(self._labels, return_inverse=True, axis=0)
+        coeff_list = np.zeros(len(self._coeffs))
+        for i, val in zip(indexes, self._coeffs):
+            coeff_list[i] += val
+        non_zero = [
+            i
+            for i, v in enumerate(coeff_list)
+            if not np.isclose(v, 0, atol=atol, rtol=rtol)
+        ]
+        label_list = label_list[non_zero]
+        coeff_list = coeff_list[non_zero]
+        if not non_zero:
+            return FermionicOp(("I" * self.register_length, 0))
+        return FermionicOp(list(zip(label_list, coeff_list)))
+
+    def __len__(self):
+        return len(self._labels)
+
+    def to_opflow(self, method: str = "JW") -> PauliSumOp:
+        # TODO: other mappings
+        # if method == "JW":
+
+        # pylint: disable=cyclic-import
+        from qiskit_nature.mappings.jordan_wigner_mapping import \
+            JordanWignerMapping
+
+        return JordanWignerMapping().map(self)
```

Comment:
See also the ParticleOp 'map' method comment below. If this is here in my mind it should be map(self, Mapper) where a given mapper instance is passed. Or at the very least if there is a shortcut string that is wanted for convenience maybe we can do that via a Mapper Factory or something. I had the impression though from the review that having a mapping function in the operator was being pushed back against.

Reply:
Thank you. I've removed `to_opflow()`. When the design of Mapper is close to completed, we may revive it, but let's put it on hold for now.
I think (self, Mapper) is very nice because it can avoid the cyclic-dependency.